### PR TITLE
refactor: use webassemblyjs tools

### PR DESCRIPTION
**Pull request by xtuc**
**What kind of change does this PR introduce?**

**Did you add tests for your changes?**

**If relevant, link to documentation update:**

**Summary**

**Does this PR introduce a breaking change?**

**Other information**

Updated to the major release.

Note that the project as been renamed to `webassemblyjs`, could you please rename the title of the PR?

### CODE REVIEW

File: lib/WebAssemblyParser.js [hunk 1]
```diff
@@ -4,7 +4,8 @@
 */
 "use strict";
 
-const Tools = require("webassembly-interpreter/lib/tools");
+const traverse = require("@webassemblyjs/ast").traverse;
+const decode = require("@webassemblyjs/wasm-parser").decode;
```
Comment (xtuc) [inline lib/WebAssemblyParser.js]: I would prefer `const {traverse} = require("@webassemblyjs/ast")` but it would have been inconsistent

Comment (michael-ciniawsky) [reply lib/WebAssemblyParser.js]: ? 🙃 Just use it and if you like update the other imports aswell :P

Comment (xtuc) [reply lib/WebAssemblyParser.js]: That looks intentional, probably because of < Node 4?

Comment (michael-ciniawsky) [reply lib/WebAssemblyParser.js]: Yep sure, but `webpack >= v4.0.0` has `node >= v6.0.0` as minimum, so the code is simply just not 'upgraded' here yet :)

File: lib/WebAssemblyParser.js [hunk 2]
```diff
@@ -4,28 +4,40 @@
 */
 "use strict";
 
-const Tools = require("webassembly-interpreter/lib/tools");
-
-const Tapable = require("tapable").Tapable;
+const {
```
Comment (EugeneHlushko) [inline lib/WebAssemblyParser.js]: is multiline required here for single variable declaration by code style configuration?

File: lib/WebAssemblyParser.js [hunk 3]
```diff
@@ -4,28 +4,40 @@
...
+	decode
+} = require("@webassemblyjs/wasm-parser");
+
+const {
+	Tapable
+} = require("tapable");
 const WebAssemblyImportDependency = require("./dependencies/WebAssemblyImportDependency");
 
+const decoderOpts = {
+	ignoreCodeSection: true,
```
Comment (EugeneHlushko) [inline lib/WebAssemblyParser.js]: is there a case when we might want to provide override for these from config?

File: lib/WebAssemblyParser.js (hunk 2 above)
Comment (xtuc) [reply lib/WebAssemblyParser.js]: yes

File: lib/WebAssemblyParser.js (hunk 3 above)
Comment (xtuc) [reply lib/WebAssemblyParser.js]: In short term probably not, we can easily expose it after since it's only in Webpack's internals.

**Comment by sokra** [comment]
Thanks

**Pull request merged**